client/src/test/java/io/pravega/client/stream/mock/MockByteStreamWriter.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.client.stream.mock;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.client.byteStream.ByteStreamWriter;
+import io.pravega.client.segment.impl.SegmentMetadataClient;
+import io.pravega.client.segment.impl.SegmentOutputStream;
+import io.pravega.client.stream.impl.PendingEvent;
+import io.pravega.common.concurrent.Futures;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+
+@RequiredArgsConstructor
+public class MockByteStreamWriter extends ByteStreamWriter {
```

Comment:
I am not clear on why this class is needed. In the tests it looks like its being created and wrapped with a `Mockito.spy` however this can be done with the real implementation just as easily.

Reply:
Removed MockByteStreamWriter, and started using ByteStreamWriterImpl instead in tests. updateLastEventFuture() is now package private and test is defined under the package.